hello-data-portal/hello-data-portal-api/src/main/java/ch/bedag/dap/hellodata/portal/initialize/service/DefaultUserInitializer.java

```diff
@@ -71,27 +71,48 @@ public class DefaultUserInitializer {
     @Transactional(propagation = Propagation.REQUIRES_NEW)
     public boolean initDefaultUsers() {
         boolean defaultUsersInitiated = false;
-        if (defaultAdminProperties.getEmail() == null) {
+        String username = defaultAdminProperties.getUsername();
+        String email = defaultAdminProperties.getEmail();
+        if (email == null) {
             log.warn("No default admin properties set, omitting");
             return defaultUsersInitiated;
         }
+        List<UserRepresentation> allUsersFromKeycloak = getAllUsersFromKeycloak();
+        Optional<UserRepresentation> userByEmail = allUsersFromKeycloak.stream().filter(user -> user.getEmail().equalsIgnoreCase(email)).findFirst();
+        Optional<UserRepresentation> userByUsername = allUsersFromKeycloak.stream().filter(user -> user.getUsername().equalsIgnoreCase(username)).findFirst();
+        boolean theSameUser = userByEmail.isPresent() && userByUsername.isPresent() && userByUsername.get().getId().equalsIgnoreCase(userByEmail.get().getId());
+
+        if (!theSameUser) {
+            log.info("Users fetched from the keycloak:");
+            allUsersFromKeycloak.forEach(
+                    userRepresentation -> log.info("Usr {}, username: {}, email: {}", userRepresentation.getId(), userRepresentation.getUsername(), userRepresentation.getEmail()));
+            throw new IllegalStateException(
+                    String.format("There are already two different users in the keycloak for the provided username: %s and the email: %s. Please change the configuration",
+                                  username, email));
+        }
+
         // Check if the default user exists in Keycloak
-        boolean userExistsInKeycloak = getAllUsersFromKeycloak().stream().anyMatch(user -> user.getEmail().equals(defaultAdminProperties.getEmail()));
+        boolean userExistsInKeycloak = allUsersFromKeycloak.stream().anyMatch(user -> user.getEmail().equals(email));
 
         // Check if the user has already been created in a previous run
-        boolean userMarkedAsDefault = !defaultUserRepository.findByEmail(defaultAdminProperties.getEmail()).isEmpty();
-
-        if (!userExistsInKeycloak && !userMarkedAsDefault) {
-            defaultUsersInitiated = createDefaultAdmin();
+        boolean userMarkedAsDefault = !defaultUserRepository.findByEmail(email).isEmpty();
+
+        //different email but duplicated username
+        if (!userByUsername.get().getEmail().equalsIgnoreCase(email)) {
+            defaultUsersInitiated = createDefaultAdmin(userByUsername.get().getUsername(), defaultAdminProperties.getFirstName(), defaultAdminProperties.getLastName(),
```

Comment:
Doesn't this create an exception upon calling createUserInKeycloak() as  this user already exists in KC?

Reply:
yes i does and it's good, now we'll have more logs to see where the problem is (most likely bad config)